app/controllers/user_annotations_controller.rb

```diff
@@ -152,6 +152,15 @@ def download_user_annotation
     send_data @data, type: 'text/plain', filename: filename, disposition: 'attachment'
   end
 
+  def persist_to_study
+    respond_to do |format|
+      # redirect back and say success
+      format.html { redirect_to user_annotations_path, notice: "User Annotation '#{@user_annotation.name}' will be added to the study. You will receive an email upon completion or error." }
+      format.json { render :index, status: :ok, location: user_annotations_path }
```

Comment:
This method call is running in the foreground - you should run it in the background by calling @user_annotation.delay.persist_to_study(current_user)